pkg/controller/cassandrarestore/cassandrarestore_controller.go

```diff
@@ -59,53 +59,54 @@ func add(mgr manager.Manager, r reconcile.Reconciler) error {
 		return err
 	}
 
-	// Watch for changes to primary resource CassandraCluster
-	err = c.Watch(&source.Kind{Type: &api.CassandraRestore{}}, &handler.EnqueueRequestForObject{},
-		predicate.Funcs{
-			CreateFunc: func(e event.CreateEvent) bool {
-				object, err := meta.Accessor(e.Object)
-				if err != nil {
+	pred := predicate.Funcs{
+		CreateFunc: func(e event.CreateEvent) bool {
+			object, err := meta.Accessor(e.Object)
+			if err != nil {
+				return false
+			}
+			if restore, ok := object.(*api.CassandraRestore); ok {
+				reqLogger := log.WithValues("Restore.Namespace", restore.Namespace, "Restore.Name", restore.Name)
+				cond := api.GetRestoreCondition(&restore.Status, api.RestoreRequired)
+				if cond != nil {
+					reqLogger.Info("Restore is already scheduled on Cluster member %s", restore.Spec.CoordinatorMember)
 					return false
 				}
-				if restore, ok := object.(*api.CassandraRestore); ok {
-					reqLogger := log.WithValues("Restore.Namespace", restore.Namespace, "Restore.Name", restore.Name)
-					cond := api.GetRestoreCondition(&restore.Status, api.RestoreRequired)
-					if cond != nil {
-						reqLogger.Info("Restore is already scheduled on Cluster member %s", restore.Spec.CoordinatorMember)
-						return false
-					}
-
-					return true
-				}
+
+				return true
+			}
+			return false
+		},
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			object, err := meta.Accessor(e.ObjectNew)
+			if err != nil {
 				return false
-			},
-			UpdateFunc: func(e event.UpdateEvent) bool {
-				object, err := meta.Accessor(e.ObjectNew)
-				if err != nil {
+			}
+			if obj, ok := object.(*api.CassandraRestore); ok {
```

Comment:
Not sure why you still need to do those tests here. You should use restore and not `obj` like you did above